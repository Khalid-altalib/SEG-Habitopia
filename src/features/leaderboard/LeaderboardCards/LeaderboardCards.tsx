import { View, VStack } from "native-base";
import React from "react";
import StatusContainer from "../../../components/StatusContainer/StatusContainer";
import LeaderboardCard from "../LeaderboardCard/LeaderboardCard";

type Props = {};

const LeaderboardCards = (props: Props) => {
  const leaderboardEntries = [
    { name: "Ihtasham", wins: 63 },
    { name: "Tom", wins: 43 },
    { name: "Bob", wins: 35 },
    { name: "Harry", wins: 29 },
    { name: "George", wins: 27 },
    { name: "George", wins: 27 },
    { name: "George", wins: 27 },
    { name: "George", wins: 27 },
    { name: "George", wins: 27 },
    { name: "George", wins: 27 },
  ];
  return (
    <StatusContainer loading={false}>
      <VStack space={5}>
        {leaderboardEntries.map((entry, index) => (
          <LeaderboardCard
            name={entry.name}
            wins={entry.wins}
            key={index}
            place={index}
          />
        ))}
      </VStack>
    </StatusContainer>
  );
};

export default LeaderboardCards;
