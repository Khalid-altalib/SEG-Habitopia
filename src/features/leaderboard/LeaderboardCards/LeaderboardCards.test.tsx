import React from "react";
import { render } from "@testing-library/react-native";
import LeaderboardCards from "./LeaderboardCards";
import TestingWrapperNavigation from "@app/testingWrapperWithNavigation";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";

/**
 * This file contains tests for the LeaderboardCards component.
 * checks if the component renders correctly with the correct data
 * checks if the component renders the correct number of cards
 */

describe("LeaderboardCards", () => {
  // Mock fetchLeaderboard action
  jest.mock("../leaderboardSlice", () => ({
    fetchLeaderboard: jest.fn(),
  }));

  const mockState = {
    leaderboard: {
      loading: false,
      error: null,
      challengeType: "Sleep",
      timeInterval: "Weekly",
      entries: [
        { name: "Alice", checkins: 4, userId: "100" },
        { name: "Bob", checkins: 3, userId: "200" },
        { name: "Charlie", checkins: 2, userId: "300" },
      ],
    },
  };
  const mockStore = configureStore([thunk])(mockState);
  let wrapper: any;

  beforeEach(() => {
    wrapper = render(
      <TestingWrapperNavigation store={mockStore}>
        <LeaderboardCards />
      </TestingWrapperNavigation>
    );
  });

  it("renders correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("loads 3 leaderboard cards", () => {
    expect(wrapper.getAllByTestId("leaderboard-card")).toHaveLength(3);
  });

  it("loads leaderboard cards with correct data", () => {
    expect(
      wrapper.getAllByTestId("leaderboard-card")[1].props.children[0]
    ).toBeDefined();
  });
});
