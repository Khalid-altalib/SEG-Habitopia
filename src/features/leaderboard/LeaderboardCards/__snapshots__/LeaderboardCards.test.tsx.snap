// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LeaderboardLayout renders correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    style={
      [
        {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <RNSScreenStack
      style={
        {
          "flex": 1,
        }
      }
    >
      <RNSScreen
        collapsable={false}
        forwardedRef={[Function]}
        gestureResponseDistance={
          {
            "bottom": -1,
            "end": -1,
            "start": -1,
            "top": -1,
          }
        }
        nativeBackButtonDismissalEnabled={false}
        onAppear={[Function]}
        onDisappear={[Function]}
        onDismissed={[Function]}
        onHeaderBackButtonClicked={[Function]}
        onNativeDismissCancelled={[Function]}
        onTransitionProgress={[Function]}
        onWillDisappear={[Function]}
        replaceAnimation="push"
        stackPresentation="push"
        style={
          {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
        swipeDirection="horizontal"
      >
        <RNSScreenStackHeaderConfig
          backButtonInCustomView={false}
          backgroundColor="rgb(255, 255, 255)"
          color="rgb(0, 122, 255)"
          direction="ltr"
          disableBackButtonMenu={false}
          hidden={false}
          hideBackButton={false}
          largeTitleHideShadow={false}
          title="TestScreen"
          titleColor="rgb(28, 28, 30)"
          topInsetEnabled={false}
          translucent={false}
        />
        <View
          accessibilityElementsHidden={false}
          importantForAccessibility="auto"
          style={
            {
              "flex": 1,
              "flexDirection": "column-reverse",
            }
          }
        >
          <View
            style={
              [
                {
                  "flex": 1,
                },
                {
                  "backgroundColor": "rgb(242, 242, 242)",
                },
                undefined,
              ]
            }
          >
            <View
              dataSet={{}}
              style={{}}
            >
              <View
                dataSet={{}}
                style={{}}
              >
                <View
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    {
                      "opacity": 1,
                    }
                  }
                  testID="leaderboard-card"
                >
                  <View
                    style={
                      {
                        "shadowColor": "black",
                        "shadowOffset": {
                          "height": 5,
                          "width": 5,
                        },
                        "shadowOpacity": 0.05,
                        "width": "100%",
                      }
                    }
                  >
                    <ViewManagerAdapter_ExpoLinearGradient
                      proxiedProperties={
                        {
                          "colors": [
                            4294688548,
                            4294759245,
                          ],
                          "endPoint": [
                            0.5,
                            2,
                          ],
                          "locations": undefined,
                          "startPoint": undefined,
                        }
                      }
                      style={
                        [
                          {
                            "borderRadius": 10,
                            "padding": 20,
                          },
                        ]
                      }
                    >
                      <View
                        dataSet={{}}
                        style={
                          {
                            "alignItems": "center",
                            "flexDirection": "row",
                            "justifyContent": "space-between",
                          }
                        }
                      >
                        <View
                          dataSet={{}}
                          style={
                            {
                              "flexDirection": "row",
                            }
                          }
                        >
                          <View
                            dataSet={{}}
                            style={
                              {
                                "height": 50,
                                "width": 50,
                              }
                            }
                          >
                            <View
                              dataSet={{}}
                              style={
                                [
                                  {},
                                  {
                                    "aspectRatio": 1,
                                    "style": undefined,
                                  },
                                ]
                              }
                            >
                              <View
                                dataSet={{}}
                                style={
                                  [
                                    {
                                      "alignItems": "center",
                                      "backgroundColor": "#a1a1aa",
                                      "borderColor": "#27272a",
                                      "borderRadius": 9999,
                                      "height": "100%",
                                      "justifyContent": "center",
                                      "position": "relative",
                                      "width": "100%",
                                    },
                                    {
                                      "bottom": 0,
                                      "left": 0,
                                      "position": "absolute",
                                      "right": 0,
                                      "top": 0,
                                    },
                                  ]
                                }
                              >
                                <Image
                                  accessibilityLabel="--"
                                  alt="--"
                                  dataSet={{}}
                                  onError={[Function]}
                                  source={
                                    {
                                      "uri": "https://robohash.org/100",
                                    }
                                  }
                                  style={
                                    [
                                      {
                                        "borderRadius": 9999,
                                        "maxWidth": "100%",
                                      },
                                      {
                                        "height": "100%",
                                        "width": "100%",
                                      },
                                    ]
                                  }
                                />
                              </View>
                            </View>
                          </View>
                          <View
                            dataSet={{}}
                            style={
                              {
                                "width": 25,
                              }
                            }
                          />
                          <View
                            dataSet={{}}
                            style={{}}
                          >
                            <Text
                              style={
                                [
                                  {
                                    "color": "#0f172a",
                                  },
                                  {
                                    "fontFamily": undefined,
                                    "fontSize": 23,
                                  },
                                  undefined,
                                ]
                              }
                            >
                              Alice
                            </Text>
                            <Text
                              style={
                                [
                                  {
                                    "color": "#0f172a",
                                  },
                                  {
                                    "fontFamily": undefined,
                                    "fontSize": 19,
                                  },
                                  undefined,
                                ]
                              }
                            >
                              4
                               Checkins
                            </Text>
                          </View>
                        </View>
                        <View
                          dataSet={{}}
                          style={{}}
                        />
                        <View
                          dataSet={{}}
                          style={{}}
                        >
                          <Text
                            style={
                              [
                                {
                                  "color": "#0f172a",
                                },
                                {
                                  "fontFamily": undefined,
                                  "fontSize": 23,
                                },
                                undefined,
                              ]
                            }
                          >
                            #
                            1
                          </Text>
                        </View>
                      </View>
                    </ViewManagerAdapter_ExpoLinearGradient>
                  </View>
                </View>
                <View
                  dataSet={{}}
                  style={
                    {
                      "height": 25,
                    }
                  }
                />
                <View
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    {
                      "opacity": 1,
                    }
                  }
                  testID="leaderboard-card"
                >
                  <View
                    style={
                      {
                        "shadowColor": "black",
                        "shadowOffset": {
                          "height": 5,
                          "width": 5,
                        },
                        "shadowOpacity": 0.05,
                        "width": "100%",
                      }
                    }
                  >
                    <ViewManagerAdapter_ExpoLinearGradient
                      proxiedProperties={
                        {
                          "colors": [
                            4294677052,
                            4294818420,
                          ],
                          "endPoint": [
                            0.5,
                            2,
                          ],
                          "locations": undefined,
                          "startPoint": undefined,
                        }
                      }
                      style={
                        [
                          {
                            "borderRadius": 10,
                            "padding": 20,
                          },
                        ]
                      }
                    >
                      <View
                        dataSet={{}}
                        style={
                          {
                            "alignItems": "center",
                            "flexDirection": "row",
                            "justifyContent": "space-between",
                          }
                        }
                      >
                        <View
                          dataSet={{}}
                          style={
                            {
                              "flexDirection": "row",
                            }
                          }
                        >
                          <View
                            dataSet={{}}
                            style={
                              {
                                "height": 50,
                                "width": 50,
                              }
                            }
                          >
                            <View
                              dataSet={{}}
                              style={
                                [
                                  {},
                                  {
                                    "aspectRatio": 1,
                                    "style": undefined,
                                  },
                                ]
                              }
                            >
                              <View
                                dataSet={{}}
                                style={
                                  [
                                    {
                                      "alignItems": "center",
                                      "backgroundColor": "#a1a1aa",
                                      "borderColor": "#27272a",
                                      "borderRadius": 9999,
                                      "height": "100%",
                                      "justifyContent": "center",
                                      "position": "relative",
                                      "width": "100%",
                                    },
                                    {
                                      "bottom": 0,
                                      "left": 0,
                                      "position": "absolute",
                                      "right": 0,
                                      "top": 0,
                                    },
                                  ]
                                }
                              >
                                <Image
                                  accessibilityLabel="--"
                                  alt="--"
                                  dataSet={{}}
                                  onError={[Function]}
                                  source={
                                    {
                                      "uri": "https://robohash.org/200",
                                    }
                                  }
                                  style={
                                    [
                                      {
                                        "borderRadius": 9999,
                                        "maxWidth": "100%",
                                      },
                                      {
                                        "height": "100%",
                                        "width": "100%",
                                      },
                                    ]
                                  }
                                />
                              </View>
                            </View>
                          </View>
                          <View
                            dataSet={{}}
                            style={
                              {
                                "width": 25,
                              }
                            }
                          />
                          <View
                            dataSet={{}}
                            style={{}}
                          >
                            <Text
                              style={
                                [
                                  {
                                    "color": "#0f172a",
                                  },
                                  {
                                    "fontFamily": undefined,
                                    "fontSize": 23,
                                  },
                                  undefined,
                                ]
                              }
                            >
                              Bob
                            </Text>
                            <Text
                              style={
                                [
                                  {
                                    "color": "#0f172a",
                                  },
                                  {
                                    "fontFamily": undefined,
                                    "fontSize": 19,
                                  },
                                  undefined,
                                ]
                              }
                            >
                              3
                               Checkins
                            </Text>
                          </View>
                        </View>
                        <View
                          dataSet={{}}
                          style={{}}
                        />
                        <View
                          dataSet={{}}
                          style={{}}
                        >
                          <Text
                            style={
                              [
                                {
                                  "color": "#0f172a",
                                },
                                {
                                  "fontFamily": undefined,
                                  "fontSize": 23,
                                },
                                undefined,
                              ]
                            }
                          >
                            #
                            2
                          </Text>
                        </View>
                      </View>
                    </ViewManagerAdapter_ExpoLinearGradient>
                  </View>
                </View>
                <View
                  dataSet={{}}
                  style={
                    {
                      "height": 25,
                    }
                  }
                />
                <View
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    {
                      "opacity": 1,
                    }
                  }
                  testID="leaderboard-card"
                >
                  <View
                    style={
                      {
                        "shadowColor": "black",
                        "shadowOffset": {
                          "height": 5,
                          "width": 5,
                        },
                        "shadowOpacity": 0.05,
                        "width": "100%",
                      }
                    }
                  >
                    <ViewManagerAdapter_ExpoLinearGradient
                      proxiedProperties={
                        {
                          "colors": [
                            4288455599,
                            4291941851,
                          ],
                          "endPoint": [
                            0.5,
                            2,
                          ],
                          "locations": undefined,
                          "startPoint": undefined,
                        }
                      }
                      style={
                        [
                          {
                            "borderRadius": 10,
                            "padding": 20,
                          },
                        ]
                      }
                    >
                      <View
                        dataSet={{}}
                        style={
                          {
                            "alignItems": "center",
                            "flexDirection": "row",
                            "justifyContent": "space-between",
                          }
                        }
                      >
                        <View
                          dataSet={{}}
                          style={
                            {
                              "flexDirection": "row",
                            }
                          }
                        >
                          <View
                            dataSet={{}}
                            style={
                              {
                                "height": 50,
                                "width": 50,
                              }
                            }
                          >
                            <View
                              dataSet={{}}
                              style={
                                [
                                  {},
                                  {
                                    "aspectRatio": 1,
                                    "style": undefined,
                                  },
                                ]
                              }
                            >
                              <View
                                dataSet={{}}
                                style={
                                  [
                                    {
                                      "alignItems": "center",
                                      "backgroundColor": "#a1a1aa",
                                      "borderColor": "#27272a",
                                      "borderRadius": 9999,
                                      "height": "100%",
                                      "justifyContent": "center",
                                      "position": "relative",
                                      "width": "100%",
                                    },
                                    {
                                      "bottom": 0,
                                      "left": 0,
                                      "position": "absolute",
                                      "right": 0,
                                      "top": 0,
                                    },
                                  ]
                                }
                              >
                                <Image
                                  accessibilityLabel="--"
                                  alt="--"
                                  dataSet={{}}
                                  onError={[Function]}
                                  source={
                                    {
                                      "uri": "https://robohash.org/300",
                                    }
                                  }
                                  style={
                                    [
                                      {
                                        "borderRadius": 9999,
                                        "maxWidth": "100%",
                                      },
                                      {
                                        "height": "100%",
                                        "width": "100%",
                                      },
                                    ]
                                  }
                                />
                              </View>
                            </View>
                          </View>
                          <View
                            dataSet={{}}
                            style={
                              {
                                "width": 25,
                              }
                            }
                          />
                          <View
                            dataSet={{}}
                            style={{}}
                          >
                            <Text
                              style={
                                [
                                  {
                                    "color": "#0f172a",
                                  },
                                  {
                                    "fontFamily": undefined,
                                    "fontSize": 23,
                                  },
                                  undefined,
                                ]
                              }
                            >
                              Charlie
                            </Text>
                            <Text
                              style={
                                [
                                  {
                                    "color": "#0f172a",
                                  },
                                  {
                                    "fontFamily": undefined,
                                    "fontSize": 19,
                                  },
                                  undefined,
                                ]
                              }
                            >
                              2
                               Checkins
                            </Text>
                          </View>
                        </View>
                        <View
                          dataSet={{}}
                          style={{}}
                        />
                        <View
                          dataSet={{}}
                          style={{}}
                        >
                          <Text
                            style={
                              [
                                {
                                  "color": "#0f172a",
                                },
                                {
                                  "fontFamily": undefined,
                                  "fontSize": 23,
                                },
                                undefined,
                              ]
                            }
                          >
                            #
                            3
                          </Text>
                        </View>
                      </View>
                    </ViewManagerAdapter_ExpoLinearGradient>
                  </View>
                </View>
              </View>
              <View
                dataSet={{}}
                style={
                  {
                    "alignItems": "center",
                    "display": "flex",
                    "justifyContent": "center",
                  }
                }
              />
            </View>
          </View>
        </View>
      </RNSScreen>
    </RNSScreenStack>
  </View>
</RNCSafeAreaProvider>
`;
